engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,10 +492,21 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog()) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
```

Comment:
If toString() is overriden it will provide more info. If it throws an exception it is handled. I think all cases are taken care of. It is better to have it in there and have a chance of getting more info.